command-expansion-server/src/lib/codegen/CommandEDSLPreface.ts

```diff
@@ -2,46 +2,126 @@
 export class Command<A extends ArgType[] | { [argName: string]: any } = [] | {}> {
   public readonly stem: string;
   public readonly arguments: A;
+  public readonly absoluteTime: Temporal.Instant | null = null;
+  public readonly epochTime: { epochName: string; time: Temporal.Duration } | null = null;
+  public readonly relativeTime: Temporal.Duration | null = null;
 
-  private constructor(opts: { stem: string; arguments: A }) {
+  private constructor(opts: { stem: string; arguments: A } & TimingType) {
```

Comment:
Can we extract this type? It seems to be duplicated inline in this file

Reply:
Hi Chris, are you talking about` { stem: string; arguments: A }` `A` is a generic value so I am not sure how to pull that part out as a type?

Reply:
```
type CommandConstructorArgs<A> = {
  stem: string;
  arguments: A;
} & ({
} | {
} | {
} | {})
```

Reply:
Something like:
```ts
type CommandOptions<A> = {
  arguments: A;
  stem: string;
} & TimingType;
```

Reply:
Jinx!

Reply:
ohhh got it. Thanks!!!

Reply:
Should be good to go. I ran through expansion without any errors.